modules/programs/zsh.nix

```
@@ -346,7 +346,11 @@ in
           fpath+="$HOME/${pluginsDir}/${plugin.name}"
         '') cfg.plugins)}
 
-        ${optionalString cfg.enableCompletion "autoload -U compinit && compinit"}
+        # Oh-My-Zsh calls compinit during initialization,
+        # calling it twice causes sight start up slowdown
+        # as all $fpath entries will be traversed again.
+        ${optionalString (cfg.enableCompletion && !cfg.oh-my-zsh.enable) "autoload -U compinit && compinit"}
```

I'd suggest splitting the line like in `enableAutosuggestions` below. Something like
```suggestion
        ${optionalString (cfg.enableCompletion && !cfg.oh-my-zsh.enable)
          "autoload -U compinit && compinit"
        }
```